lib/get-esm-exports.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+
+const { Parser } = require('acorn')
+const { importAssertions } = require('acorn-import-assertions');
+
+const acornOpts = {
+  ecmaVersion: 'latest',
+  sourceType: 'module'
+}
+
+const parser = Parser.extend(importAssertions)
+
+function getEsmExports (moduleStr) {
+  const exportedNames = new Set()
+  const tree = parser.parse(moduleStr, acornOpts)
+  for (const node of tree.body) {
+    if (!node.type.startsWith('Export')) continue
+    switch (node.type) {
+      case 'ExportNamedDeclaration':
+        if (node.declaration) {
+          parseDeclaration(node, exportedNames)
+        } else {
+          parseSpecifiers(node, exportedNames)
+        }
+        break
+      case 'ExportDefaultDeclaration':
+        exportedNames.add('default')
+        break
+      case 'ExportAllDeclaration':
+        if (node.exported) {
+          exportedNames.add(node.exported.name)
+        } else {
+          exportedNames.add('*')
+        }
+        break
+      default:
+        throw new Error('unrecognized export type: ' + node.type)
```

Comment:
I do wonder, if node adds a new export type at some point, is it safer to warn and ignore it?

Reply:
Actually, this is probably tied to an enumeration of values built into this specific version of Acorn and therefor isn't likely to change.

Reply:
It'll change if:

1. There's an AST type that can go here that we're not yet testing for.
2. An AST type is added as the language evolves.

Ignoring is probably the safest in terms of not breaking user code. A warning might help indicate what the issue is.